### ISSUE TITLE
Doc Question: Temp Storage Cleanup

### ISSUE THREAD

**Issue by Strepto** [Azure/azure-storage-fuse#654]
From readme.md <https://github.com/Azure/azure-storage-fuse#performance-and-caching> 

> Blobfuse currently does not manage available disk space in the tmp path. Make sure to have enough space, or reduce --file-cache-timeout-in-seconds value to accelerating purging cached files.

But this MountOptions parameter exists <https://github.com/Azure/azure-storage-fuse#mount-options>

> [OPTIONAL] --high-disk-threshold=90 : High disk threshold percentage. When the disk usage reaches this threshold cache eviction resumes. This parameter overrides 'file-cache-timeout-in-seconds' parameter and cached files will be removed even if it is not expired. Files which are currently in use (open) will not be evicted from cache.

Is one of the docs outdated or are both correct? Is there anything I am missing here?

**Comment by vibhansa-msft** [comment]
@Strepto : These are two different things.
--file-cache-timeout : just means for how long file can be kept on the disk before it can be purged/evicted from cache.
--high-threshold : means if the disk usage reaches a certain % then do not wait for cache timeout and start evicting older files which are closed. However this % is caculated based on the actual disk and not just the temp directory usage. So for e.g. if your temp directory is /home/abc/temp then % usage is for "/"

These are two different parameters to control your local disk usage. They control when the eviction policy is kicked in,

**Comment by vibhansa-msft** [comment]
Understood your point but what I am trying to convey is blobfuse does not manage disk space for the temp path, that % is for the mount point of that temp path. So lets say if your disk is already 70% full then after mounting blobfuse as soon as you put 10% more load in temp path, it will start evicting and lower threshold will never hit because even after deleting files from temp path disk usage is never going to go down to 60% of its capacity. 
So in short it tries to manage but not in absolute terms.

**Comment by Strepto** [comment]
Oh, I misunderstood the docs then.

Blob fuse manages its own Cache in absolute disk space terms. So when I add Fuse to a cache disk with 50% use from another app. Fuse says that the cache is 50% full already. But Fuse cannot delete this data, as its not written by Fuse?

So Fuse manages its own downloaded data, and after it reaches 90% of total disk storage it will remove until it has 80% remaining. 
If the other app increases its cache to 90% of the entire disk, Fuse will not be able to clean anything and might end up in a constant cleaning state.

Do I understand correctly now?

If so I would consider re-writing the message below in the docs since everyone in my team thought this meant that we had to tweak the `--file-cache-timeout-in-seconds` to a number that would never hit 100% disk storage. We understood it as there was no disk space threshold for deleting cache before the timeout.

>Blobfuse currently does not manage available disk space in the tmp path. Make sure to have enough space, or reduce --file-cache-timeout-in-seconds value to accelerating purging cached files.

**Comment by vibhansa-msft** [comment]
--file-cache-timeout-in-seconds : seconds for which files downloaded by blobfuse are considered valid by blobfuse. So value of this shall depend on your application use-case. If you set it 0 then as soon as you close the file, blobfuse will delete it.

For the threshold part you understood it correctly, its total disk space and not just the temp folder. So if there are files other then what blobfuse downloaded occupying the disk space, blobfuse will never be able to delete them as its only aware of the files it downloaded.

**Comment by vibhansa-msft** [comment]
"--file-cache-timeout-in-seconds to a number that would never hit 100% disk storage" is not correct because this is per file timeout. If application opens huge number of files in parallel then all will occupy disk space and it may hit 100% capacity as well. Other scenario is if you open a very huge file which can consume all of remaining disk space, you will hit 100%. So just setting cache-timeout to a smaller value does not ensure anything. Its just a indication on when a particular file will be deleted.

"We understood it as there was no disk space threshold for deleting cache before the timeout." : High and low thresholds are for this only, only catch there is while calculating the disk usage it considers the mount point and not just the folder that you mentioned. If total disk usage has reached threshold then blobfuse will start deleting the files before timeout, but it can only delete what it has downloaded.
